qiskit_experiments/library/characterization/analysis/multi_state_discrimination_analysis.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Multi state discrimination analysis."""
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+from typing import List, Tuple
+
+from qiskit.providers.options import Options
+from qiskit_experiments.framework import BaseAnalysis, AnalysisResultData, ExperimentData
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from qiskit_experiments.data_processing import SkCLF
+
+from sklearn.inspection import DecisionBoundaryDisplay
+from sklearn.linear_model import SGDClassifier
+
+
+class MultiStateDiscriminationAnalysis(BaseAnalysis):
+    """This class will fit the discriminator to the data."""
```

Comment:
```suggestion
    """This class fits a multi-state discriminator to the data."""
```

Reply:
Addressed in commit [9390381](https://github.com/Qiskit/qiskit-experiments/pull/936/commits/9390381ce44f62e19b8243b76bd212b37df0fa8f).